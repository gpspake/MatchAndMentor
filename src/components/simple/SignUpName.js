import React from 'react';
import {
  View,
  Text,
  TextInput,
  Keyboard,
  TouchableWithoutFeedback,
  Platform,
  KeyboardAvoidingView,
} from 'react-native';
import Button from './Button';
import ScreenHeader from './ScreenHeader';
import { submitName, updateName } from '../../actions';

const styles = {
  container: {
    flex: 1,
    backgroundColor: '#fff',
    justifyContent: 'space-between',
  },
  buttonStyle: {
    backgroundColor: '#55A838',
    margin: 20,
    height: 45,
    borderRadius: 10,
    justifyContent: 'center',
  },
  textStyle: {
    color: 'white',
    alignSelf: 'center',
    fontSize: 24,
    fontFamily: 'Montserrat-Regular',
  },
  inputStyle: {
    height: 50,
    borderBottomColor: 'black',
    borderBottomWidth: Platform.OS === 'ios' ? 1 : 0,
    margin: 20,
    fontSize: 24,
    fontFamily: 'Montserrat-Regular',
  },
};

function submit(name, dispatch) {
  Keyboard.dismiss();
  dispatch(submitName(name));
}
function SignUpName(props) {
  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
      <View style={styles.container}>
        <View>
          <ScreenHeader title={props.title} />
          <TextInput
            style={styles.inputStyle}
            onChangeText={value => props.dispatch(updateName(value))}
            maxLength={100}
            onSubmitEditing={Keyboard.dismiss}
            onEndEditing={Keyboard.dismiss}
            testID="name-input"
          />
        </View>
        <KeyboardAvoidingView behavior="padding" keyboardVerticalOffset={64}>
          <Button
            style={
              props.name !== null && props.name !== ''
                ? styles.buttonStyle
                : { ...styles.buttonStyle, backgroundColor: '#e9e9e9' }
            }
            onPress={() => submit(props.name, props.dispatch)}
            disabled={props.name === null || props.name === ''}
            testID="name-submit-button">
            <Text style={styles.textStyle}>Continue</Text>
          </Button>
        </KeyboardAvoidingView>
      </View>
    </TouchableWithoutFeedback>
  );
}

export default SignUpName;
