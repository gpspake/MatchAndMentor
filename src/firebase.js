import firebase from 'firebase';

// Initialize Firebase
const config = {
  apiKey: 'AIzaSyAsjC_abpWmFCYmNoQl257nc807hHE_gsY',
  authDomain: 'matchandmentorapp.firebaseapp.com',
  databaseURL: 'https://matchandmentorapp.firebaseio.com',
  projectId: 'matchandmentorapp',
  storageBucket: 'matchandmentorapp.appspot.com',
  messagingSenderId: '1068797055216',
};
firebase.initializeApp(config);

export default firebase;

export const database = firebase.database();
export const auth = firebase.auth();
export const googleAuthProvider = new firebase.auth.GoogleAuthProvider();
