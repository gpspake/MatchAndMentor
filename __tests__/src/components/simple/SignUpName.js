/* eslint-env jest */
'use strict'
import { Keyboard } from 'react-native'
import React from 'react'
import SignUpName from '../../../../src/components/simple/SignUpName'
import renderer from 'react-test-renderer'
import { findById } from '../../../../setupTests'
import firebase from 'firebase'

if (!firebase.apps.length) {
  firebase.initializeApp({
    apiKey: 'AIzaSyAsjC_abpWmFCYmNoQl257nc807hHE_gsY',
    authDomain: 'matchandmentorapp.firebaseapp.com',
    databaseURL: 'https://matchandmentorapp.firebaseio.com',
    projectId: 'matchandmentorapp',
    storageBucket: 'matchandmentorapp.appspot.com',
    messagingSenderId: '1068797055216'
  })
}

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

jest.mock('Keyboard', () => {
  return {
    dismiss: jest.fn(),
    addListener: jest.fn()
  }
})

const tree = renderer.create(<SignUpName dispatch={jest.fn()} name='test' />)

it('renders correctly', () => {
  expect(tree).toMatchSnapshot()
})

it('calls submit on button press', () => {
  const submitButton = findById(tree.toJSON(), 'name-submit-button')
  submitButton.props.onPress()
  expect(Keyboard.dismiss).toBeCalled()
})

it('calls dispatch on change text', () => {
  const nameInput = findById(tree.toJSON(), 'name-input')
  nameInput.props.onChangeText()
  expect(tree.toTree().props.dispatch).toBeCalled()
})
