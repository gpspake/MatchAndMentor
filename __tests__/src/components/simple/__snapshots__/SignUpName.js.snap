// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
  testID={undefined}
>
  <View>
    <View
      style={
        Object {
          "margin": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "black",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 28,
          }
        }
      />
    </View>
    <TextInput
      maxLength={100}
      onChangeText={[Function]}
      onEndEditing={[Function]}
      onSubmitEditing={[Function]}
      style={
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "fontFamily": "Montserrat-Regular",
          "fontSize": 24,
          "height": 50,
          "margin": 20,
        }
      }
      testID="name-input"
    />
  </View>
  <View
    keyboardVerticalOffset={64}
    onLayout={[Function]}
    style={
      Array [
        undefined,
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      disabled={false}
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "#55A838",
          "borderRadius": 10,
          "height": 45,
          "justifyContent": "center",
          "margin": 20,
        }
      }
      testID="name-submit-button"
      underlayColor="black"
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontFamily": "Montserrat-Regular",
            "fontSize": 24,
          }
        }
      >
        Continue
      </Text>
    </TouchableHighlight>
  </View>
</View>
`;
